src/hera/workflows/_meta_mixins.py

```diff
@@ -531,10 +531,16 @@ def _create_subnode(
         from hera.workflows.task import Task
         from hera.workflows.workflow_template import WorkflowTemplate
 
+        subnode_name = subnode_name.replace("_", "-")
```

Comment:
nit: should we maybe do this when we fetch it via `varname` instead of here? Replacing something explicitly provided by the user may be surprising, even if what they gave was wrong.